docarray/array/array.py

```diff
@@ -381,3 +384,22 @@ def traverse_flat(
         flattened = AnyDocumentArray._flatten_one_level(nodes)
 
         return flattened
+
+    @overload
+    def __delitem__(self: T, key: int) -> None:
+        ...
+
+    @overload
+    def __delitem__(self: T, key: IndexIterType) -> None:
+        ...
+
+    def __delitem__(self, key) -> None:
+        key = self._normalize_index_item(key)
+
+        if key is None:
+            return
```

Comment:
what is happening when you do `return` alone ?

Reply:
returns None